### PR TITLE
Add linters for ROS1

### PR DESCRIPTION
**Pull request by ojasjoshi**
*Issue #, if available:*

*Description of changes:*
- Added linters for ROS1
- Fixed linting errors.
- Added .py extensions to rospy nodes. Changed their references in the required launch files.

*Tests*
- Tested locally on Melodic+G9 that the build succeeds and the all launch files work as expected. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ament_lint:
+    runs-on: ubuntu-latest
+    container:
+      image: rostooling/setup-ros-docker:ubuntu-bionic-ros-eloquent-ros-base-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        linter: [copyright, flake8, pep257, xmllint]
+    steps:
+      - run: sudo chown -R rosbuild:rosbuild "$HOME" .
```
Comment (mm318) [inline .github/workflows/lint.yml]: We shouldn't need this anymore since https://github.com/ros-tooling/setup-ros-docker/issues/7 has been resolved.

**Comment by ahtsan** [comment]
If only `.py` files are checked, are we missing most of the files for ROS1? like all the `.launch` files

**Comment by ahtsan** [comment]
> If only `.py` files are checked, are we missing most of the files for ROS1? like all the `.launch` files

Looking at the logs, I only see `.py` files are checked, e.g. under flake8
```
 2 files checked
No problems found

Checked files:

* robot_ws/src/hello_world_robot/setup.py
* robot_ws/src/hello_world_robot/src/hello_world_robot/__init__.py
```

**Comment by mm318** [comment]
> If only `.py` files are checked, are we missing most of the files for ROS1? like all the `.launch` files

We are also missing files like [this](https://github.com/aws-robotics/aws-robomaker-sample-application-helloworld/blob/ros1/robot_ws/src/hello_world_robot/nodes/rotate), because they don't have a `.py` extension even though they are Python scripts.

**Comment by ojasjoshi** [comment]
> > If only `.py` files are checked, are we missing most of the files for ROS1? like all the `.launch` files
> 
> Looking at the logs, I only see `.py` files are checked, e.g. under flake8
> 
> ```
>  2 files checked
> No problems found
> 
> Checked files:
> 
> * robot_ws/src/hello_world_robot/setup.py
> * robot_ws/src/hello_world_robot/src/hello_world_robot/__init__.py
> ```

Hmm, good catch, xmllint isn't processing launch files as an xml.

**Comment by ojasjoshi** [comment]
> > If only `.py` files are checked, are we missing most of the files for ROS1? like all the `.launch` files
> 
> We are also missing files like [this](https://github.com/aws-robotics/aws-robomaker-sample-application-helloworld/blob/ros1/robot_ws/src/hello_world_robot/nodes/rotate), because they don't have a `.py` extension even though they are Python scripts.

Is having these python files with no extension a best practice? I think adding a .py extension is the solution here.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (ojasjoshi) [reply .github/workflows/lint.yml]: Ah nice, finally! We should update the rest of the workflows as well

**Review by mm318 — APPROVED** [review]

**Comment by mm318** [comment]
> Is having these python files with no extension a best practice? I think adding a .py extension is the solution here.

Yeah, I agree with putting back the `.py` extension.

I'm not aware of excluding the extension being a good practice. The examples in the Python ROS Publisher/Subscriber tutorial do keep their extensions (http://wiki.ros.org/rospy_tutorials/Tutorials/WritingPublisherSubscriber).

**Pull request merged**